### PR TITLE
Add more platform support for IsDebuggerAttached

### PR DESCRIPTION
**Pull request by SNSTRUTHERS**
Adds proper breakpoint support when a debugger is attached to the engine process for macOS and Linux.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,8 @@
 *.sln
 *.vcxproj
 *.vcxproj.user
-*.vcxproj.filters
\ No newline at end of file
+*.vcxproj.filters
+
+Makefile
+*.exe
```
Comment (anthofoxo) [inline .gitignore:11]: unnecessary, build files are placed within /bin

File: game/src/engine/log.cpp [hunk 2]
```diff
@@ -58,6 +67,50 @@ namespace FoxEngine
...
+		mib[3] = getpid();
+		len = sizeof kp;
+
+		iError = sysctl(mib, 4, &kp, &len, nullptr, 0);
+		if (iError) return false;
+
+		return kp.kp_proc.p_flag & P_TRACED;
 #else
 		// Assume no debugger is attached
 		return false;
```
Comment (anthofoxo) [inline game/src/engine/log.cpp:116]: Probably should put a #warn here

**Review by anthofoxo — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (SNSTRUTHERS) [reply .gitignore:11]: Then why was there a `game/game.exe` in one of the commits?

Comment (anthofoxo) [reply .gitignore:11]: Well somebody forgot to delete it (it was me) oopsies

Comment (SNSTRUTHERS) [reply .gitignore:11]: Hence the addition to the `.gitignore` lol

Comment (anthofoxo) [reply .gitignore:11]: fair enough

File: game/src/engine/log.cpp [hunk 3]
```diff
@@ -112,6 +112,7 @@ namespace FoxEngine
 
 		return kp.kp_proc.p_flag & P_TRACED;
 #else
+#	pragma message "Warning: platform cannot determine whether debugger is present; assumes that debugger is never attached"
```
Comment (anthofoxo) [inline game/src/engine/log.cpp:115]: apparently #warning is platform specific but this will work ^^;

**Review by anthofoxo — APPROVED** [review]

**Pull request merged**